File: C++/Questions/Leaf-similar.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void inOrder(TreeNode* root, string& s){
        if(root == NULL) return;
        if(root -> right == NULL && root -> left == NULL){
            s += to_string(root->val)+ "_";
            return ;
        }
        inOrder(root->left, s);
        inOrder(root->right, s);
    }
    bool leafSimilar(TreeNode* root1, TreeNode* root2) {
        string s1 = "";
        string s2 = "";

        inOrder(root1, s1);
        inOrder(root2, s2);

        if(s1 == s2){
            return true;
        }
        else return false;
        
    }
};

File: Python/Questions/Leaf-Similar.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def inOrder(self, root, s):
        if not root:
            return
        if not root.left and not root.right:
            s.append(str(root.val) + "_")
            return
        self.inOrder(root.left, s)
        self.inOrder(root.right, s)
    
    def leafSimilar(self, root1, root2):
        s1 = []
        s2 = []

        self.inOrder(root1, s1)
        self.inOrder(root2, s2)